File: Python/HashTable/BJAlgo_2002.py
from sys import stdin

N = int(stdin.readline())
result = 0

# 들어간 차량 순서대로 저장 {'차량번호': 입장순서}
cars = {}
for i in range(N):
    car = stdin.readline().strip('\n')
    cars[car] = i

for i in range(N):
    car = stdin.readline().strip('\n')
    num = cars[car]  # 들어온 차량 찾아서
    del cars[car]    # dictionary 에서 삭제 후
    for val in cars.values():  # 남은 차량의 입장 순서를 확인해
        if num > val:          # 현재 차량보다 빨리 들어온 차가 있는 경우 result + 1
            result += 1
            break

print(result)

File: Java/HashTable/BJAlgo_2002.java
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.Map;

public class BJAlgo_2002 {
    public static void main(String[] args) throws Exception{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int result = 0;
        int N = Integer.parseInt(br.readLine());

        Map<String, Integer> cars = new HashMap<>();
        for(int i = 0; i < N; i++) {
            String car = br.readLine();
            cars.put(car, i);
        }

        for(int i = 0; i < N; i++) {
            String car = br.readLine();
            int num = cars.get(car);
            cars.remove(car);
            for(int j : cars.values()) {
                if (num > j) {
                    result += 1;
                    break;
                }
            }
        }

        System.out.println(result);
    }
}
